src/main/java/net/neoforged/neoforge/transfer/handlers/templates/resources/InfiniteResourceHandler.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.templates.resources;
+
+import java.util.Objects;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.handlers.resources.ResourceHandler;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.resources.ResourceStack;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+
+/**
+ * A {@link ResourceHandler} that allows extraction of an unlimited amount of a specified resource.
+ *
+ * @param <T> The type of resource that this storage can accept.
+ */
+// TODO: questionable usefulness
+public class InfiniteResourceHandler<T extends IResource> implements ResourceHandler<T> {
```

Comment:
I feel like `Infinate` and `Voiding` could be the same implementations. `ConstantResourceHandler(IResource, allowInsert, allowExtract)` perhaps?

Reply:
They are similar to some extent, but I think that trying to share some code will be more hurtful than anything. The differences include:
- One has an empty resource, the other not.
- One accepts all insertions, the other only of the infinite resource.
- One does not allow extractions, the other does only of the infinite resource.
- One has a 0 amount, the other `Long.MAX_VALUE`.
- `isValid` implementations are different as well.

Reply:
Fair enough, I was considering a use case of a creative tank, or barrel. Where someone would want to have a consistent stock of resource, but still allow voiding of excess on insert. The idea being if you wanted truly void, you just make one with an empty resource and you get full void semantics.

However, could be added later should it actually come up in prod.